File: ios/Sources/VideoRecorder/TrimSettings.swift
import Foundation

enum TrimSettingsError: Error {
    case invalidArgument(message: String)
}

/// Trim boundaries of an edit operation, expressed in milliseconds.
public class TrimSettings: NSObject {
    private var startsAt: CLong = 0
    private var endsAt: CLong = 0

    /// - Parameters:
    ///   - startsAt: start of the output, in milliseconds from the start of the source
    ///   - endsAt: end of the output in milliseconds, or 0 for "until the end of the source"
    init(startsAt: CLong, endsAt: CLong) throws {
        super.init()

        try self.setStartsAt(startsAt)
        try self.setEndsAt(endsAt)
    }

    /// Get startsAt in milliseconds
    func getStartsAt() -> CLong {
        return self.startsAt
    }

    /// Set startsAt in milliseconds
    func setStartsAt(_ startsAt: CLong) throws {
        if startsAt < 0 {
            throw TrimSettingsError.invalidArgument(message: "Parameter startsAt cannot be negative")
        }
        self.startsAt = startsAt
    }

    /// Get endsAt in milliseconds. A value of 0 means "until the end of the source".
    func getEndsAt() -> CLong {
        return self.endsAt
    }

    /// Set endsAt in milliseconds
    func setEndsAt(_ endsAt: CLong) throws {
        if endsAt < 0 {
            throw TrimSettingsError.invalidArgument(message: "Parameter endsAt cannot be negative")
        }
        self.endsAt = endsAt
    }
}

File: android/src/main/java/com/capacitorcommunity/videorecorder/editor/TrimSettings.java
package com.capacitorcommunity.videorecorder.editor;

/**
 * Trim boundaries of an edit operation, expressed in milliseconds.
 */
public class TrimSettings {

    private long startsAt = 0;
    private long endsAt = 0;

    public TrimSettings() {}

    public TrimSettings(long startsAt, long endsAt) {
        setStartsAt(startsAt);
        setEndsAt(endsAt);
    }

    /**
     * Get startsAt in milliseconds
     *
     * @return startsAt in milliseconds
     */
    public long getStartsAt() {
        return startsAt;
    }

    public void setStartsAt(long startsAt) {
        if (startsAt < 0) {
            throw new IllegalArgumentException("Parameter startsAt cannot be negative");
        }

        this.startsAt = startsAt;
    }

    /**
     * Get endsAt in milliseconds. A value of 0 means "until the end of the source".
     *
     * @return endsAt in milliseconds
     */
    public long getEndsAt() {
        return endsAt;
    }

    public void setEndsAt(long endsAt) {
        if (endsAt < 0) {
            throw new IllegalArgumentException("Parameter endsAt cannot be negative");
        }

        this.endsAt = endsAt;
    }
}
